Utils/Services/ImagePickerService.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  ImagePickerService.swift
+//  Utils
+//
+//  Created by Nahuel Gladstein on 6/13/17.
+//  Copyright © 2017 Wolox. All rights reserved.
+//
+
+import ReactiveSwift
+import UIKit
+import AVFoundation
+import Photos
+import Result
+
+public enum ImagePickerServiceError: Error {
+    
+    case sourceTypeNotAvailable
+    
+}
+
+public protocol ImagePickerServiceType {
```

Comment:
The thing is that the protocol doesn't have an init with a controller, so it's weird that it can present a controller with only a `permissionNotGranted ` closure as information